### PR TITLE
Break up b2.py

### PR DESCRIPTION
**Pull request by bwbeach**

### CODE REVIEW

File: b2/account_info.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+######################################################################
+
+import json
+import os
+import stat
+from abc import (ABCMeta, abstractmethod)
+
+import six
+
+from .exception import (MissingAccountData)
```
Comment (ppolewicz) [inline b2/account_info.py]: unnecessary parenthesis

File: b2/account_info.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+# Copyright 2016 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+import json
+import os
+import stat
+from abc import (ABCMeta, abstractmethod)
```
Comment (ppolewicz) [inline b2/account_info.py]: unnecessary parenthesis

File: b2/account_info.py [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+        flags = os.O_WRONLY | os.O_CREAT | os.O_TRUNC
+        if os.name == 'nt':
+            flags |= os.O_BINARY
+        with os.fdopen(os.open(self.filename, flags, stat.S_IRUSR | stat.S_IWUSR), 'wb') as f:
+            # is there a cleaner way to do this that works in both Python 2 and 3?
+            json_bytes = json.dumps(self.data, indent=4, sort_keys=True).encode('utf-8')
+            f.write(json_bytes)
+
+
+class StubAccountInfo(AbstractAccountInfo):
```
Comment (ppolewicz) [inline b2/account_info.py]: I think this should be kept somewhere in `tests/` - we don't want to expose it to the user next to `StoredAccountInfo`

File: b2/api.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+#
+# File: b2/api.py
+#
+# Copyright 2016 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from .account_info import (StoredAccountInfo)
```
Comment (ppolewicz) [inline b2/api.py]: unnecessary parenthesis here and below

File: b2/account_info.py (hunk 1 above)
Comment (bwbeach) [inline b2/account_info.py]: Yes.  They're unnecessary.  I'm undecided whether I like the consistency of always using parens on `from ... import ...` or not.  If you don't like them, I'm OK with not using them.

I was getting tired of adding and removing them as the import lists got bigger and smaller.

Comment (bwbeach) [inline b2/account_info.py]: I think yapf adds parens when it goes multi-line, and doesn't remove them when i shrinks.

File: b2/download_dest.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+import os
+from abc import (ABCMeta, abstractmethod)
+
+import six
+
+from .progress import (StreamWithProgress)
+
+
+@six.add_metaclass(ABCMeta)
+class DownloadDest(object):
```
Comment (ppolewicz) [inline b2/download_dest.py]: I suggest to rename it to `AbstractDownloadDestination`

File: b2/raw_api.py [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+            error_dict = json.loads(e.data.decode('utf-8'))
+            raise map_error_dict_to_exception(e, error_dict, params)
+        except ValueError:
+            v = sys.exc_info()
+            error_dict = {'error_decoding_json': v}
+            raise FatalError('error decoding JSON when handling an exception', [e_backup, v],)
+
+
+@six.add_metaclass(ABCMeta)
+class RawApi(object):
```
Comment (ppolewicz) [inline b2/raw_api.py]: I suggest to rename to `AbstractRawApi` for consistency with other abstract classes that we have

File: b2/sync.py [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+
+import os
+import time
+from abc import (ABCMeta, abstractmethod)
+
+import six
+
+
+@six.add_metaclass(ABCMeta)
+class Action(object):
```
Comment (ppolewicz) [inline b2/sync.py]: `AbstractAction`

File: b2/sync.py [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+
+    def latest_version(self):
+        return self.versions[0]
+
+    def __repr__(self):
+        return 'File(%s, [%s])' % (self.name, ', '.join(map(repr, self.versions)))
+
+
+@six.add_metaclass(ABCMeta)
+class Folder(object):
```
Comment (ppolewicz) [inline b2/sync.py]: `AbstractFolder`

**Pull request merged**